### PR TITLE
Feature/upgrade to kafka v3

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

Trello card: https://trello.com/c/OOVFfUxv/1225-1225-upgrade-search-data-extractor-dp-kafka-to-v3

- Upgrade dp-kafka dependency to version 3
- Upgrade code and tests accordingly

Depends on this [dataset api pr](https://github.com/ONSdigital/dp-dataset-api/pull/412)

### How to review

- Make sure code changes make sense. You may use the following [repo](https://github.com/ONSdigital/dp-cantabular-csv-exporter) as a reference to validate the structure.
- Make sure unit tests pass
- Make sure component tests pass

### Who can review

Anyone

### CODE REVIEW

File: features/steps/steps.go [hunk 1]
```diff
@@ -143,26 +141,10 @@ func (c *Component) thisSearchDataImportEventIsSent(eventDocstring *godog.DocStr
...
-		}
-
-		var e = &models.SearchDataImport{}
-		if err := schema.SearchDataImportEvent.Unmarshal(msg, e); err != nil {
-			return fmt.Errorf("error unmarshalling message: %w", err)
-		}
-
-		return fmt.Errorf("kafka event received in csv-created topic: %v", e)
-	}
+	return c.KafkaProducer.WaitNoMessageSent(c.waitEventTimeout)
```
Comment (redhug1) [inline features/steps/steps.go:147]: The WaitNoMessageSent function is not using time.After correctly, please update it as per the notes i have sent you directly in Slack.
Thanks

**Review by redhug1 — CHANGES_REQUESTED** [review]
see single comment

File: clients/datasetapi.go [hunk 2]
```diff
@@ -12,5 +12,6 @@ import (
 // DatasetApiClient defines the zebedee client
 type DatasetClient interface {
 	Checker(context.Context, *healthcheck.CheckState) error
+	GetDatasetCurrentAndNext(ctx context.Context, userAuthToken, serviceAuthToken, collectionID, datasetID string) (m datasetclient.Dataset, err error)
```
Comment (nshumoogum) [inline clients/datasetapi.go:15]: We shouldn't need this method once we add the additional metadata to the metadata endpoint

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: clients/datasetapi.go (hunk 2 above)
Comment (DavidSubiros) [reply clients/datasetapi.go]: Add dependency with dataset api PR, and removed latest commit

**Review by nshumoogum — APPROVED** [review]
Looks good to me

**Review by redhug1 — APPROVED** [review]

**Pull request merged**